api/v2beta2/helmrelease_types.go

```diff
@@ -0,0 +1,1036 @@
+	if in.Timeout == nil {
+		return defaultTimeout
+	}
+	return *in.Timeout
+}
+
+// Rollback holds the configuration for Helm rollback actions for this
+// HelmRelease.
+type Rollback struct {
+	// Timeout is the time to wait for any individual Kubernetes operation (like
+	// Jobs for hooks) during the performance of a Helm rollback action. Defaults to
+	// 'HelmReleaseSpec.Timeout'.
+	// +optional
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+	// DisableWait disables the waiting for resources to be ready after a Helm
+	// rollback has been performed.
+	// +optional
+	DisableWait bool `json:"disableWait,omitempty"`
+
+	// DisableWaitForJobs disables waiting for jobs to complete after a Helm
+	// rollback has been performed.
+	// +optional
+	DisableWaitForJobs bool `json:"disableWaitForJobs,omitempty"`
+
+	// DisableHooks prevents hooks from running during the Helm rollback action.
+	// +optional
+	DisableHooks bool `json:"disableHooks,omitempty"`
+
+	// Recreate performs pod restarts for the resource if applicable.
+	// +optional
+	Recreate bool `json:"recreate,omitempty"`
+
+	// Force forces resource updates through a replacement strategy.
+	// +optional
+	Force bool `json:"force,omitempty"`
+
+	// CleanupOnFail allows deletion of new resources created during the Helm
+	// rollback action when it fails.
+	// +optional
+	CleanupOnFail bool `json:"cleanupOnFail,omitempty"`
+}
+
+// GetTimeout returns the configured timeout for the Helm rollback action, or
+// the given default.
+func (in Rollback) GetTimeout(defaultTimeout metav1.Duration) metav1.Duration {
+	if in.Timeout == nil {
+		return defaultTimeout
+	}
+	return *in.Timeout
+}
+
+// Uninstall holds the configuration for Helm uninstall actions for this
+// HelmRelease.
+type Uninstall struct {
+	// Timeout is the time to wait for any individual Kubernetes operation (like
+	// Jobs for hooks) during the performance of a Helm uninstall action. Defaults
+	// to 'HelmReleaseSpec.Timeout'.
+	// +optional
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+	// DisableHooks prevents hooks from running during the Helm rollback action.
+	// +optional
+	DisableHooks bool `json:"disableHooks,omitempty"`
+
+	// KeepHistory tells Helm to remove all associated resources and mark the
+	// release as deleted, but retain the release history.
+	// +optional
+	KeepHistory bool `json:"keepHistory,omitempty"`
+
+	// DisableWait disables waiting for all the resources to be deleted after
+	// a Helm uninstall is performed.
+	// +optional
+	DisableWait bool `json:"disableWait,omitempty"`
+}
+
+// GetTimeout returns the configured timeout for the Helm uninstall action, or
+// the given default.
+func (in Uninstall) GetTimeout(defaultTimeout metav1.Duration) metav1.Duration {
+	if in.Timeout == nil {
+		return defaultTimeout
+	}
+	return *in.Timeout
+}
+
+// HelmReleaseInfo holds the status information for a Helm release as performed
+// by the controller.
+type HelmReleaseInfo struct {
```

Comment:
Maybe it'd be helpful for this to have a `StorageNamespace` field as well.

Reply:
This will be added when the logic for changes to the `targetNamespace` and/or `storageNamespace` is added to the release, which includes e.g. garbage collection.